tests/snuba/api/endpoints/test_organization_events_vitals.py

```diff
@@ -0,0 +1,164 @@
+        url = reverse(
+            "sentry-api-0-organization-events-vitals",
+            kwargs={"organization_slug": self.organization.slug},
+        )
+
+        with self.feature(features):
+            return self.client.get(url, query, format="json")
+
+    def test_no_projects(self):
+        response = self.do_request()
+        assert response.status_code == 200, response.content
+        assert len(response.data) == 0
+
+    def test_no_vitals(self):
+        self.store_event(
+            self.transaction_data, project_id=self.project.id,
+        )
+
+        self.query.update({"vital": []})
+        response = self.do_request()
+        assert response.status_code == 400, response.content
+        assert "Need to pass at least one vital" == response.data["detail"]
+
+    def test_bad_vital(self):
+        self.store_event(
+            self.transaction_data, project_id=self.project.id,
+        )
+
+        self.query.update({"vital": ["foobar"]})
+        response = self.do_request()
+        assert response.status_code == 400, response.content
+        assert "foobar is not a valid vital" == response.data["detail"]
+
+    def test_simple(self):
+        data = self.transaction_data.copy()
+        for lcp in [2000, 3000, 5000]:
+            self.store_event(
+                data, {"lcp": lcp}, project_id=self.project.id,
+            )
+
+        self.query.update({"vital": ["measurements.lcp"]})
+        response = self.do_request()
+        assert response.status_code == 200, response.content
+        assert response.data["measurements.lcp"] == {
+            "group_0": 1,
+            "group_1": 1,
+            "group_2": 1,
+            "total": 3,
+            "p75": 4000,
+        }
+
+    def test_grouping(self):
+        counts = [
+            (100, 2),
+            (3000, 3),
+            (4500, 1),
+        ]
+        for duration, count in counts:
+            for _ in range(count):
+                self.store_event(
+                    load_data("transaction", timestamp=self.start),
+                    {"lcp": duration},
+                    project_id=self.project.id,
+                )
+
+        self.query.update({"vital": ["measurements.lcp"]})
+        response = self.do_request()
+        assert response.status_code == 200
+        print(response.data)
+        assert response.data["measurements.lcp"] == {
+            "group_0": 2,
+            "group_1": 3,
+            "group_2": 1,
+            "total": 6,
+            "p75": 3000,
+        }
+
+    def test_multiple_vitals(self):
+        vitals = {"lcp": 3000, "fcp": 5000}
+        self.store_event(
+            load_data("transaction", timestamp=self.start), vitals, project_id=self.project.id,
+        )
+
+        self.query.update({"vital": ["measurements.lcp", "measurements.fcp"]})
+        response = self.do_request()
+        assert response.status_code == 200
+        assert response.data["measurements.lcp"] == {
```

Comment:
Should we test all of them here since we are partially testing in the other tests?

Reply:
Ah yeah :+1: makes sense